adserver/models.py

```diff
@@ -291,33 +291,48 @@ def daily_reports(
 
             days[index]["date"] = impression.date
             days[index]["index"] = index_display
+            days[index]["decisions"] += impression.decisions
+            days[index]["offers"] += impression.offers
             days[index]["views"] += impression.views
             days[index]["clicks"] += impression.clicks
-            days[index]["revenue"] += (
-                impression.clicks * float(impression.advertisement.flight.cpc)
-            ) + (impression.views * float(impression.advertisement.flight.cpm) / 1000.0)
-            days[index]["revenue_share"] = days[index]["revenue"] * (
-                self.revenue_share_percentage / 100.0
-            )
-            days[index]["our_revenue"] = (
-                days[index]["revenue"] - days[index]["revenue_share"]
-            )
+
+            if impression.advertisement:
+                days[index]["revenue"] += (
```

Comment:
Should we set these to 0 in the `else` case?

Reply:
We're doing a `+=`, so setting it to 0 will break things. I think this logic works -- at least it works fine in my testing. I guess it could break for a site with only Decisions and no Offers/Views